images/airflow/2.10.3/python/mwaa/celery/sqs_broker.py

```diff
@@ -603,7 +605,12 @@ def basic_publish(self, message, exchange, routing_key, **kwargs):
         if os.environ.get('MWAA__HEALTH_MONITORING_ENABLE_REVAMPED_HEALTHCHECK', 'false') == 'true' and exchange == 'celeryev':
             # This branch catches all celery task events and generates process heartbeat metrics
             if routing_key == 'worker.heartbeat':
-                Stats.incr("mwaa.celery.process.heartbeat", 1)
+                Stats.gauge("mwaa.celery.process.heartbeat", 1)
+                num_active_tasks = len(self._get_tasks_from_state(self.celery_state)) if self.idle_worker_monitoring_enabled else CELERY_WORKER_TASK_LIMIT
+                if num_active_tasks >= CELERY_WORKER_TASK_LIMIT:
```

Comment:
in case `MWAA__CORE__TASK_MONITORING_ENABLED` is `false` we would be emitting the value as the limit, wouldnt this be a wrong indication?